x/staging/posts/keeper/grpc_query.go

```diff
@@ -12,102 +11,47 @@ import (
 	"google.golang.org/grpc/status"
 
 	"github.com/desmos-labs/desmos/x/staging/posts/types"
+	subspacestypes "github.com/desmos-labs/desmos/x/staging/subspaces/types"
 )
 
 var _ types.QueryServer = Keeper{}
 
-func (k Keeper) getPostResponse(ctx sdk.Context, post types.Post) types.QueryPostResponse {
-	// Get the reactions
-	postReactions := k.GetPostReactions(ctx, post.PostID)
-	if postReactions == nil {
-		postReactions = []types.PostReaction{}
-	}
-
-	// Get the children
-	childrenIDs := k.GetPostChildrenIDs(ctx, post.PostID)
-	if childrenIDs == nil {
-		childrenIDs = []string{}
-	}
-
-	//Get the user answers if poll exist
-	var answers []types.UserAnswer
-	if post.PollData != nil {
-		answers = k.GetUserAnswersByPost(ctx, post.PostID)
-	}
-
-	// Crete the response object
-	return types.QueryPostResponse{
-		Post:        post,
-		UserAnswers: answers,
-		Reactions:   postReactions,
-		Children:    childrenIDs,
-	}
-}
-
+// Posts implements the Query/Posts gRPC method
 func (k Keeper) Posts(goCtx context.Context, req *types.QueryPostsRequest) (*types.QueryPostsResponse, error) {
-	var filteredPosts []types.QueryPostResponse
+	var posts []types.Post
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	if !subspacestypes.IsValidSubspace(req.SubspaceId) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace must be a valid sha-256 hash")
```

Comment:
I think we can simplify this maybe using `subspacetypes.ErrInvalidSubspaceID` instead: 

```golang
		return nil, sdkerrors.Wrapf(subspacetypes.ErrInvalidSubspaceID, req.SubspaceId)
```

If this variable is not present, we should create it